### PR TITLE
`improvement` Enable library evolution

### PR DESCRIPTION
**Pull request by goergisn**
## New
<new>

Support for [library evolution](https://www.swift.org/blog/library-evolution/)

</new>

## Summary
- Adds default implementation for protocol `@_spi` properties
- Adds tests to guarantee `assertionFailure` on unimplemented protocol conformance
- Fixes unrelated warnings
- Adds a [github workflow](https://github.com/Adyen/adyen-ios/actions/workflows/test_library_evolution.yml) that ensures the compatibility with `BUILD_LIBRARY_FOR_DISTRIBUTION=YES`

See: https://github.com/Adyen/adyen-ios/issues/1354

### CODE REVIEW

File: Tests/Adyen Tests/Analytics/AnalyticsProviderTests.swift [hunk 1]
```diff
@@ -255,7 +254,7 @@ class AnalyticsProviderTests: XCTestCase {
         )
         
         let encodedRequest = try JSONEncoder().encode(request)
-        var decodedRequest = try XCTUnwrap(JSONSerialization.jsonObject(with: encodedRequest) as? [String: Any])
+        let decodedRequest = try XCTUnwrap(JSONSerialization.jsonObject(with: encodedRequest) as? [String: Any])
```
Comment (goergisn) [inline Tests/Adyen Tests/Analytics/AnalyticsProviderTests.swift]: Unrelated change but fixes a warning

File: Tests/Adyen Tests/Analytics/AnalyticsProviderTests.swift [hunk 2]
```diff
@@ -163,7 +163,6 @@ class AnalyticsProviderTests: XCTestCase {
     func testTelemetryRequest() throws {
         // Given
         
-        let amount = Amount(value: 1, currencyCode: "EUR")
```
Comment (goergisn) [inline Tests/Adyen Tests/Analytics/AnalyticsProviderTests.swift]: Unrelated change but fixes a warning

File: Adyen/Core/APIClient/APIContext.swift [hunk 3]
```diff
@@ -48,3 +48,35 @@ public struct APIContext: AnyAPIContext {
...
+    
+    /// This default implementation has to be provided to be able to build with `BUILD_LIBRARY_FOR_DISTRIBUTION` enabled
+    ///
+    /// - Warning: Access will cause an failure in debug mode to assure the correct implementation of the `AdyenContextAware` protocol
+    @_spi(AdyenInternal)
+    var context: AdyenContext {
+        get {
+            AdyenAssertion.assertionFailure(
+                message: "`@_spi(AdyenInternal) var context: AdyenContext` needs to be provided on `\(String(describing: Self.self))`"
+            )
```
Comment (descorp) [inline Adyen/Core/APIClient/APIContext.swift]: What if one make debug run with `BUILD_LIBRARY_FOR_DISTRIBUTION` ?

Comment (goergisn) [reply Adyen/Core/APIClient/APIContext.swift]: It only fails if the "default implementation" is accessed which should never be the case if `AdyenContextAware` is implemented correctly.

So doing a debug run with `BUILD_LIBRARY_FOR_DISTRIBUTION` won't be affected by that

**Review by erenbesel — APPROVED** [review]

**Pull request merged**